### DISCUSSION TITLE
Use LiveWire with Sourceguardian encoder

### DISCUSSION THREAD

**Discussion by NKazantsev** [livewire/livewire#3906] [1. Help]
Hi, we test https://www.sourceguardian.com for hardcode our project.
All works fine, include blades encoding (for prepare blades we use this script https://www.sourceguardian.com/knowledgebase/laravel/precompileblade.phps )
But if we use livewire in blades (@livewire() or < livewire /> directives) - all brokens after transform blades.
Is anybody encode projects with Sourceguardian and livewire ?

**Comment by joshhanley** [comment]
@NKazantsev the reason that script probably won't work is because Livewire extends the blade compiler for `@livewire` to work. But that script is only using the standard blade compiler, hence your Livewire directives aren't getting compiled.

If you want to see how Livewire extends the blade compiler, have a look at https://github.com/livewire/livewire/blob/master/src/LivewireServiceProvider.php

I personally don't see the need for encoding your project, but if you really do need it, you will need to investigate how to modify that script to work with Livewire.

Hope this helps and good luck!
